core/src/main/java/org/infinispan/reactive/publisher/impl/NonSegmentedLocalPublisherManagerImpl.java

```diff
@@ -0,0 +1,74 @@
+package org.infinispan.reactive.publisher.impl;
+
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+import org.infinispan.CacheSet;
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.distribution.ch.KeyPartitioner;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.reactive.RxJavaInterop;
+import org.reactivestreams.Publisher;
+
+import io.reactivex.Flowable;
+
+/**
+ * LocalPublisherManager that handles cases when a non segmented store are present. In this case we optimize
+ * the retrieval to not use individual segments as this would cause multiple full scans of the underlying
+ * store. In this case we submit a task for all segments requested and process the results concurrently if
+ * requested.
+ * @author wburns
+ * @since 10.0
+ */
+@Scope(Scopes.NAMED_CACHE)
+public class NonSegmentedLocalPublisherManagerImpl<K, V> extends LocalPublisherManagerImpl<K, V> {
+   @Inject KeyPartitioner keyPartitioner;
+
+   @Override
+   protected <I, R> Flowable<R> exactlyOnceParallel(CacheSet<I> set,
+         Set<K> keysToExclude, Function<I, K> toKeyFunction, IntSet segments,
+         Function<? super Publisher<I>, ? extends CompletionStage<R>> transformer,
+         SegmentListener listener, IntSet concurrentSegments) {
+      Flowable<I> flowable = Flowable.fromPublisher(set.localPublisher(segments));
+
+      if (keysToExclude != null) {
+         flowable = flowable.filter(i -> !keysToExclude.contains(toKeyFunction.apply(i)));
+      }
+
+      return combineStages(flowable.buffer(512)
+            .parallel()
+            .runOn(asyncScheduler)
+            .map(buffer -> transformer.apply(Flowable.fromIterable(buffer)))
```

Comment:
Can we not get a buffer that is already a `Flowable` so that we don't have to call `fromIterable`?

Reply:
I can look at the API, but I don't believe so. Either way it would have to be queued up in some fashion.